src/main/java/de/tum/in/www1/artemis/service/CourseService.java

```diff
@@ -242,8 +243,7 @@ public Course findOneWithExercisesAndLecturesAndExamsAndLearningGoalsAndTutorial
      * @return an unmodifiable list of all courses for the user
      */
     public List<Course> findAllActiveForUser(User user) {
-        return courseRepository.findAllActive(ZonedDateTime.now()).stream().filter(course -> course.getEndDate() == null || course.getEndDate().isAfter(ZonedDateTime.now()))
-                .filter(course -> isCourseVisibleForUser(user, course)).toList();
+        return courseRepository.findAllActive(ZonedDateTime.now()).stream().filter(course -> isCourseVisibleForUser(user, course)).toList();
```

Comment:
Just to understand this line of code correctly: We step away from the current behaviour to hide courses after the end date, correct?

Reply:
`findAllActive` already does this filtering, there is no need to do it again in Java